iscan/static/iscan/query-detail/query-detail.js

```diff
@@ -31,415 +31,485 @@ angular.module('queryDetail', [
             });
         }
     };
-}])
-    .controller('QueryDetailCtrl', function ($scope, Query, Corpora, $state, $stateParams, $document, Annotations, djangoAuth, Users, $mdDialog) {
-            $scope.paginateParams = Query.paginateParams;
-            $scope.annotation_types = Query.annotation_types;
-            $scope.newAnnotation = {};
-            $scope.currentAnnotations = {};
-            $scope.headline = 'Loading detail...';
-            $scope.selection_begin = 0;
-            $scope.selection_end = null;
-            $scope.selection_anchor = null;
-            $scope.detail_index = parseInt($stateParams.detail_index);
-
-
-        $scope.refreshPermissions = function () {
-            if ($scope.user.is_superuser) {
-                $scope.can_view = true;
-            }
-            else {
+}]).controller('QueryDetailCtrl', function ($scope, Query, Corpora, $state, $stateParams, $document, Annotations, djangoAuth, Users, $mdDialog) {
+    $scope.paginateParams = Query.paginateParams;
+    $scope.annotation_types = Query.annotation_types;
+    $scope.newAnnotation = {};
+    $scope.currentAnnotations = {};
+    $scope.headline = 'Loading detail...';
+    $scope.typing = false;
+    $scope.has_edited_subannotations == false;
+    $scope.selected_subannotation = '';
+    $scope.selection_begin = 0;
+    $scope.selection_end = null;
+    $scope.selection_anchor = null;
+    $scope.detail_index = parseInt($stateParams.detail_index);
+
+    $scope.refreshPermissions = function () {
+        if ($scope.user.is_superuser) {
+            $scope.can_view = true;
+        }
+        else {
 
-                $scope.can_view = false;
-                for (i = 0; i < $scope.user.corpus_permissions.length; i++) {
-                    if ($scope.user.corpus_permissions[i].corpus == $stateParams.corpus_id) {
-                        $scope.can_view = $scope.user.corpus_permissions[i].can_view_detail;
-                    }
+            $scope.can_view = false;
+            for (i = 0; i < $scope.user.corpus_permissions.length; i++) {
+                if ($scope.user.corpus_permissions[i].corpus == $stateParams.corpus_id) {
+                    $scope.can_view = $scope.user.corpus_permissions[i].can_view_detail;
```

Comment:
So here I would add the flag for `$scope.can_edit` which then will disable moving of boundaries or changing any fields values.